nixos/modules/services/torrent/transmission.nix

```diff
@@ -104,47 +99,79 @@ in
   };
 
   config = mkIf cfg.enable {
-    systemd.tmpfiles.rules = [
-      "d '${homeDir}' 0770 '${cfg.user}' '${cfg.group}' - -"
-      "d '${settingsDir}' 0700 '${cfg.user}' '${cfg.group}' - -"
-      "d '${fullSettings.download-dir}' '${downloadDirPermissions}' '${cfg.user}' '${cfg.group}' - -"
-      "d '${fullSettings.incomplete-dir}' '${downloadDirPermissions}' '${cfg.user}' '${cfg.group}' - -"
+    systemd.tmpfiles.rules =
+      optionals (! hasPrefix "/var/lib/" cfg.home) [
+        "d '${cfg.home}' 755 '${cfg.user}' '${cfg.group}' - -"
+        # Avoid "Unsafe path transition" by specifying parent directory .config
+        "d '${cfg.home}/.config' 755 '${cfg.user}' '${cfg.group}' - -"
+        "d '${cfg.home}/${settingsDir}' 700 '${cfg.user}' '${cfg.group}' - -"
+      ] ++
```

Comment:
As much as like this PR and I think that the previous version of this module was awful, the permissions part is the one that has bothered me the most since always.

Thankfully, the permissions now are a little bit more configurable, but e.g you are hardcoding the permissions for the `settingsDir` and the `home` directory. It's most unfortunate that `tmpfiles.d` is not a setting a user can override via `mkForce`, in case they'd like to change these permissions. Anyway, perhaps the permissions for these directories as well should be configurable?

I once [talked on discourse](https://discourse.nixos.org/t/nixpkgs-policy-as-for-systemd-prestart-setup-scripts-vs-systemd-tmpfiles/5839) about how `systemd.tmpfiles` is bad because it's an array and not an attrset. I intend to open an RFC specifically for tmpfiles at the near future.

Reply:
That's also resolved.